src/components/Hero/Hero.tsx

```diff
@@ -1,57 +1,154 @@
-export interface HeroProps {
-  sectionIdx?: number
-  logoUrl?: string
-  heroTitle?: string
-  heroCaption?: string
+import { HeroDropdown, type HeroDropdownProps } from "./HeroDropdown"
+import {
+  HeroInfoboxDesktop,
+  HeroInfoboxMobile,
+  HeroInfoboxTablet,
+  type HeroInfoboxProps,
+} from "./HeroInfobox"
+
+export interface HeroHeadingProps {
+  title: string
+  subtitle?: string
+}
+
+export interface HeroButtonProps {
   buttonLabel?: string
   buttonUrl?: string
-  bgUrl?: string
 }
 
-export default function Hero({
-  heroTitle,
-  heroCaption,
+interface HeroKeyHighlightProps {
+  title: string
+  description: string
+  url: string
+}
+
+interface HeroCommonProps {
+  backgroundUrl: string
+  keyHighlights?: Array<HeroKeyHighlightProps>
+}
+
+export interface HeroSideProps extends HeroInfoboxProps, HeroCommonProps {
+  variant: "side"
+}
+
+interface HeroImageProps extends HeroCommonProps {
+  variant: "image"
+  dropdown?: HeroDropdownProps
+}
+
+export interface HeroFloatingProps extends HeroInfoboxProps, HeroCommonProps {
+  variant: "floating"
+}
+
+interface HeroCenterProps
+  extends HeroHeadingProps,
+    HeroButtonProps,
+    HeroCommonProps {
+  variant: "center"
+  dropdown?: HeroDropdownProps
+}
+
+export type HeroProps =
+  | HeroSideProps
+  | HeroImageProps
+  | HeroFloatingProps
+  | HeroCenterProps
+
+const BP_BUTTON_CLASSES =
+  "rounded-none box-content appearance-none items-center border border-solid border-[#f0f0f0] shadow-none inline-flex text-base h-9 justify-center px-3 py-[calc(0.375rem-1px)] relative align-top select-none cursor-pointer text-center whitespace-nowrap focus:outline-none active:outline-none disabled:cursor-not-allowed"
+
+const HeroSide = (props: Omit<HeroSideProps, keyof HeroCommonProps>) => {
+  return (
+    <div className="grow shrink-0">
+      {/* Desktop view 768 above */}
+      <div className="invisible md:visible hidden md:block">
+        <HeroInfoboxDesktop {...props} />
+      </div>
+
+      {/* Mobile view below 768 */}
+      <div className="visible md:invisible block md:hidden">
+        <HeroInfoboxMobile {...props} />
+      </div>
+    </div>
+  )
+}
+
+const HeroImage = ({
+  dropdown,
+}: Omit<HeroImageProps, keyof HeroCommonProps>) => {
+  return (
+    <div className="grow shrink-0 px-6 py-12">
+      <div className="mx-auto mt-8 mb-0">
+        <div className="flex min-h-[398px] w-11/12 md:w-5/6 lg:w-2/3 mx-auto items-center justify-center">
+          {dropdown && (
+            <div className="block basis-0 grow shrink p-3 max-w-full">
+              <HeroDropdown {...dropdown} />
+            </div>
+          )}
+        </div>
+      </div>
+    </div>
+  )
+}
+
+const HeroFloating = (
+  props: Omit<HeroFloatingProps, keyof HeroCommonProps>,
+) => {
+  return (
+    <div className="md:p-12">
+      {/* Desktop view 1024 above */}
+      <div className="invisible lg:visible hidden lg:block max-w-screen-xl mx-auto my-0">
```

Comment:
first time seeing `invisible`, went to look it up
can I ask whats the rationale for using it here instead of hidden?
https://tailwindcss.com/docs/visibility#making-elements-invisible

I would have thought we wouldn't want it to appear in the DOM (not sure why it isn't affecting layout currently tbh)

Reply:
Hmm for some reason I learnt before that we should specify `visibility: hidden` together with `display: none` to support screen readers, but doing a search on this topic did not yield anything and in fact, screen readers are able to detect if something is hidden using the `display: none` property. Adjusted in 214340f70037ea61e777450e9572320a4aebf48c!